### DISCUSSION TITLE
WebAssembly (WASM) support

### DISCUSSION THREAD

**Discussion by limcheekin** [ggml-org/ggml#223] [Ideas]
Hi there,

Is there any plan to support convert any GGML models (.bin) to Web Assembly (.wasm)? I think it is great idea as we can run the inference at offline mode even in the browser.

As far as I know, the whisper.cpp support Wasm and also the MNIST example (https://github.com/ggerganov/ggml/pull/190)

What do you think? I'd love to hear from you! :)

**Comment by matthoffner** [comment]
I was able to get the santacoder model to build with the command documented in the MNIST example but I haven't gotten around to testing it. I've seen that the web-llm project now supports these models which would provide WebGPU support.

**Reply by limcheekin** [reply]
Great! Appreciate if you could share your test results here.

Thanks.
